crates/protogen/proto/metastore/catalog.proto

```diff
@@ -199,9 +199,10 @@ message FunctionEntry {
     LOCAL = 1;
     REMOTE = 2;
   }
+
   EntryMeta meta = 1;
   FunctionType func_type = 2;
-  RuntimePreference runtime_preference = 3;
+  reserved 3;
```

Comment:
since this isn't public do we need to reserve this? is there any harm in just removing?

Reply:
This is tricky since these protos also have definitions that get written to disk, and so we have to be careful to make sure we're not clobbering something accidentally.

In this case, `FunctionEntry` is only sent over the network and never persisted, so removing the reserve would be fine. But it's on the programmer to know that this specific entry isn't serialized, but something like `TunnelEntry` is. Since we don't have a clear distinction, I think always being safe and reserving ids is the preferred action.

(Also when we reserve, we should make a comment on what it was previously just to provide context when we look back at it)